### PR TITLE
feat: switch to term/ansi for text manipulation

### PR DESCRIPTION
**Pull request by aymanbagabas**
Use ANSI aware, wide characters support, uniseg backed term/ansi package to calculate string widths, truncate, and wrap strings.

Related: https://github.com/muesli/reflow/pull/71
Fixes: https://github.com/charmbracelet/lipgloss/issues/258
Related: https://github.com/charmbracelet/lipgloss/issues/220

### CODE REVIEW

File: join.go [hunk 1]
```diff
@@ -85,7 +85,8 @@ func JoinHorizontal(pos Position, strs ...string) string {
 			b.WriteString(block[i])
 
 			// Also make lines the same length
-			b.WriteString(strings.Repeat(" ", maxWidths[j]-ansi.PrintableRuneWidth(block[i])))
+			// b.WriteString(strings.Repeat(" ", maxWidths[j]-ansi.StringWidth(block[i])))
```
Comment (mikelorant) [inline join.go:88]: What is the reasoning for this? Confused about the TODO as well. Is this just to remind yourself to update it?

**Review by mikelorant — COMMENTED** [review]
Everything looks good except the confusion caused by the commented out line.

File: join.go (hunk 1 above)
Comment (aymanbagabas) [reply join.go]: Hey @mikelorant, thank you for reviewing this PR. This indeed was just to remind me to update the line I was testing. I just updated the branch to uncomment this line.

**Review by mikelorant — APPROVED** [review]

File: join.go (hunk 1 above)
Comment (mikelorant) [reply join.go]: 👍

File: get.go [hunk 2]
```diff
@@ -489,10 +489,11 @@ func (s Style) getAsTransform(k propKey) func(string) string {
 // Split a string into lines, additionally returning the size of the widest
 // line.
 func getLines(s string) (lines []string, widest int) {
+	s = strings.ReplaceAll(s, "\t", "    ")
```
Comment (aymanbagabas) [inline get.go:492]:
```suggestion
```

**Review by meowgorithm — APPROVED** [review]

**Comment by mikelorant** [comment]
@aymanbagabas We need Bubbles to be updated ASAP as this will cause major inconsistencies.

**Pull request merged**